ipykernel/inprocess/client.py

```diff
@@ -15,11 +15,7 @@
 
 from jupyter_client.client import KernelClient
 from jupyter_client.clientabc import KernelClientABC
-
-try:
-    from jupyter_client.utils import run_sync  # requires 7.0+
-except ImportError:
-    run_sync = None  # type:ignore
```

Comment:
According to the inline comment here, you need to bump the minimal required version of Jupyter-client in `pyproject.toml` to be `>=7.0`.

Reply:
No, we're pulling this function from `jupyter_core` now, which I added to the deps.

Reply:
Ok, I see 👍🏽 Sorry for the noise then.